oggm/sandbox/run_alaska.py

```diff
@@ -127,65 +130,201 @@
 
-# Try many C's (they should make sense in comparison to the size of the glacier)
-c_candidates = np.linspace(0, 0.1, 10)
-for c in c_candidates:
+# Defining a calving function
+def calving_from_depth(gdir):
+    """ Finds a calving flux based on the approaches proposed by
+        Huss and Hock, (2015) and Oerlemans and Nick (2005).
 
-    # First guess calving
-    gdir.inversion_calving_rate = c
+    We take the initial output of the model and surface elevation data
+    to calculate the water depth of the calving front.
+    """
+    # We read the model output, of the last pixel of the inversion
+    cl = gdir.read_pickle('inversion_output', div_id=1)[-1]
+    t_altitude = cl['hgt'][-1] #this gives us the altitude at the terminus
+    width = cl['width'][-1]
+    thick = cl['volume'][-1] / cl['dx'] / width
+
+    # We calculate the water_depth
+    w_depth = thick - t_altitude
+    #or F_calving = np.amax(0, 2*H_free*w_depth) * width
+    print('t_altitude', t_altitude)
+    print('depth', w_depth)
+    print('thick',thick)
+
+    return np.absolute(((2*thick*w_depth)*width)/1e9), w_depth, thick
+
+#Starting the calving loop, with a maximum of 50 cycles
+i = 0
+data_calving = []
+w_depth = []
+thick = []
+
+while i < 50:
+    #TRUE
+    #First calculates a calving flux from model output with no calving
+    F_calving, new_depth, new_thick = calving_from_depth(gdir)
+
+    #Stores the data, and we see it
+    data_calving += [F_calving]
+    w_depth += [new_depth]
+    thick += [new_thick]
+    print('Calving rate calculated', F_calving)
+
+    #We put the calving function output into the Model
+    gdir.inversion_calving_rate = F_calving
 
-    # Recompute mu
+    #Recompute mu with calving
     tasks.distribute_t_stars([gdir])
 
-    # Inversion
+    # Inversion with calving
     tasks.prepare_for_inversion(gdir)
     tasks.volume_inversion(gdir)
+    tasks.distribute_thickness(gdir, how='per_altitude')
+
+    i += 1
+    print('At the bottom i is', i)
+    #calculating depth averages, to compare them, the loop is cut when it meets
+    #the condition in the if statement
+    avg_one = np.average(data_calving[-4:])
+    avg_two = np.average(data_calving[-5:-1])
+    difference = abs(avg_two - avg_one)
+    if difference < 0.05*avg_two and i>5:
+        break
+
+#This is not really needed is just for you to see whats going on
+print('All calving', data_calving)
+print('All water depth', w_depth)
+print('avg one', avg_one)
+print('avg two', avg_two)
+print('difference avg', difference)
+print('All Hi', thick)
+
+#We write everything
+gdir.write_pickle(data_calving, 'calving_fluxes', div_id=1)
+gdir.write_pickle(w_depth, 'water_depth', div_id=1)
+gdir.write_pickle(thick, 'H_ice', div_id=1)
+
+#We read everything
+all_calving_data = gdir.read_pickle('calving_fluxes', div_id=1)
+all_data_depth = gdir.read_pickle('water_depth', div_id=1)
+all_data_H_i = gdir.read_pickle('H_ice', div_id=1)
+
```

Comment:
If you change the entry in `cfg.BASENAMES`, these six lines would translate into (not tested):

```python
#We write everything
calvout = {'calving_fluxes':data_calving, 'water_depth':w_depth, 'h_ice':thick}
gdir.write_pickle(calvout, 'calving')

#We read everything
calin = gdir.read_pickle('calving')
all_calving_data = calin['calving_fluxes']
all_data_depth = calin['water_depth']
all_data_H_i = calin['H_ice']
```
Note that I also removed the ``div_id=1`` keyword, as I hope that calving glaciers won't need any divide.